### PR TITLE
Upgrade to 0.3 of the `time` crate

### PR DESCRIPTION
**Pull request by jszwedko**
Fixes: #56

Ended up pulling in a bunch of `rustfmt` changes as well.

Signed-off-by: Jesse Szwedko <jesse@szwedko.me>

### CODE REVIEW

File: src/format.rs [hunk 1]
```diff
@@ -1,129 +1,167 @@
...
-        write!(w, "<{}>{} {} {}[{}]: {}",
-          encode_priority(severity, self.facility),
-          time::now().strftime("%b %d %T").unwrap(),
-          hostname, self.process, self.pid, message).chain_err(|| ErrorKind::Format)
-    } else {
-        write!(w, "<{}>{} {}[{}]: {}",
-          encode_priority(severity, self.facility),
-          time::now().strftime("%b %d %T").unwrap(),
-          self.process, self.pid, message).chain_err(|| ErrorKind::Format)
+    fn format<W: Write>(&self, w: &mut W, severity: Severity, message: T) -> Result<()> {
```
Comment (jszwedko) [inline src/format.rs:68]: This method is what changed.

**Comment by jszwedko** [comment]
Good call @Roguelazer . I somehow misread the fix before. I pushed f09e505 which uses UTC when on unix systems.

File: src/format.rs [hunk 2]
```diff
@@ -1,129 +1,179 @@
...
+        log_message: (i32, StructuredData, T),
+    ) -> Result<()> {
+        let (message_id, data, message) = log_message;
+
+        write!(
+            w,
+            "<{}> {} {} {} {} {} {} {} {}",
+            encode_priority(severity, self.facility),
+            1, // version
+            time::OffsetDateTime::now_utc()
```
Comment (Roguelazer) [inline src/format.rs:148]: I think this also needs to call `now_local`

**Comment by Roguelazer** [comment]
@Geal is this crate still maintained?

**Comment by Geal** [comment]
@Roguelazer it has not moved much recently because I focused on other topics, but that issue with the time crate is important enough to make a new release

**Comment by Geal** [comment]
thanks for the ping though :smiley:

**Comment by Geal** [comment]
it is now rebased and merged as part of https://github.com/Geal/rust-syslog/commit/7086f02074a01c48c74135ec3d198e2205229479

hank you for the help, this will be released soon in a major version

**Pull request closed without merging**